Integrated run.py commands with flask-script and moved it to manage.py

RaghavPro opened the pull request:
Details mentioned in #109

anandology commented on manage.py:
```diff
@@ -9,8 +7,64 @@
+
+    from cleansweep.loaddata import main
+    root_dir = raw_input("Directory (Input without quotations): ")
+    main(root_dir)
```
No raw_input. It should accept arguments from command-line.

RaghavPro replied:
I don't think it allows more than one argument. You yourself mentioned in the issue. I didn't see anything in documentation but I'll look.

I used classes because dash "-" in function name isn't a valid.
I'll change it to underscore.
________________________________
From: Anand Chitipothu<mailto:notifications@github.com>
Sent: ‎22/‎09/‎2015 06:59
To: AamAadmiParty/cleansweep<mailto:cleansweep@noreply.github.com>
Cc: Raghav Sharma<mailto:Raghav_ftw@hotmail.com>
Subject: Re: [cleansweep] Integrated run.py commands with flask-script and moved it to manage.py (#120)

>
> +@manager.command
> +def init():
> +    "Initiates the application for first time use"
> +
> +    from cleansweep.loaddata import init
> +    init()
> +
> +
> +@manager.command
> +def load():
> +    "Loads data from specified directory"
> +
> +    from cleansweep.loaddata import main
> +    root_dir = raw_input("Directory (Input without quotations): ")
> +    main(root_dir)

No raw_input. It should accept arguments from command-line.

---
Reply to this email directly or view it on GitHub:
https://github.com/AamAadmiParty/cleansweep/pull/120/files#r40046598

RaghavPro replied:
Ok never mind. There are many ways to do it.
Reject this. I'll send a new request.
________________________________
From: Anand Chitipothu<mailto:notifications@github.com>
Sent: ‎22/‎09/‎2015 06:59
To: AamAadmiParty/cleansweep<mailto:cleansweep@noreply.github.com>
Cc: Raghav Sharma<mailto:Raghav_ftw@hotmail.com>
Subject: Re: [cleansweep] Integrated run.py commands with flask-script and moved it to manage.py (#120)

>
> +@manager.command
> +def init():
> +    "Initiates the application for first time use"
> +
> +    from cleansweep.loaddata import init
> +    init()
> +
> +
> +@manager.command
> +def load():
> +    "Loads data from specified directory"
> +
> +    from cleansweep.loaddata import main
> +    root_dir = raw_input("Directory (Input without quotations): ")
> +    main(root_dir)

No raw_input. It should accept arguments from command-line.

---
Reply to this email directly or view it on GitHub:
https://github.com/AamAadmiParty/cleansweep/pull/120/files#r40046598

anandology commented on manage.py:
```diff
@@ -9,8 +7,64 @@
+
+if __name__ == '__main__':
+    manager.run({'db': MigrateCommand, 'load-files': LoadFiles(), 'add-member': AddMember(), 'run-worker': RunWorker(),
+                 'update-parents': UpdateParents()})
```
Please use `@manager.command` decorator instead of creating list of commands like this. 

Also, I suggest making each command a function with decorator, instead of a class.

anandology commented:
There is no way to provide multiple arguments with the same name. It is not possible to provide multiple filenames. Thats why we've changed the `--load-files` command to `load-file`. IIRC, it can accept multiple arguments as long as they are of different names.

RaghavPro commented:
Yes. You've to pre define the arguments names.
How about we use raw_input just for load-files?
________________________________
From: Anand Chitipothu<mailto:notifications@github.com>
Sent: ‎22/‎09/‎2015 10:09
To: AamAadmiParty/cleansweep<mailto:cleansweep@noreply.github.com>
Cc: Raghav Sharma<mailto:Raghav_ftw@hotmail.com>
Subject: Re: [cleansweep] Integrated run.py commands with flask-script and moved it to manage.py (#120)

There is no way to provide multiple arguments with the same name. It is not possible to provide multiple filenames. Thats why we've changed the `--load-files` command to `load-file`. IIRC, it can accept multiple arguments as long as they are of different names.

---
Reply to this email directly or view it on GitHub:
https://github.com/AamAadmiParty/cleansweep/pull/120#issuecomment-142180307

anandology commented:
prompting for input is considered a good style in unix world. Passing command-line arguments makes it easier to use the commands in other scripts. 

Since flask-script doesn't support multiple arguments with same name, I've changed it to load-file (load single file) from load-files (load multiple files at once).
